galaxy_ng/app/migrations/0019_pulp_container_access_policies.py

```diff
@@ -0,0 +1,194 @@
+                "condition": "has_model_or_obj_perms:container.view_containernamespace",
+            },
+            {
+                "action": ["destroy"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": [
+                    "has_model_or_obj_perms:container.delete_containernamespace",
+                    "has_model_or_obj_perms:container.view_containernamespace",
+                ],
+            },
+            {
+                "action": ["create_distribution"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:container.namespace_add_containerdistribution",
+            },
+            {
+                "action": ["view_distribution"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:container.namespace_view_containerdistribution",  # noqa: E501
+            },
+        ],
+        # Removed group creation for owner. Filtering out the container groups proved to be too
+        # much of a challenge.
+        "permissions_assignment": [
+            {
+                "function": "add_for_object_creator",
+                "parameters": None,
+                "permissions": [
+                    "container.view_containernamespace",
+                    "container.delete_containernamespace",
+                    # Add `container.change_containernamespace` permissions so the namespace
+                    # owner can add additional groups to their namespace.
+                    "container.change_containernamespace",
+                    "container.namespace_add_containerdistribution",
+                    "container.namespace_delete_containerdistribution",
+                    "container.namespace_view_containerdistribution",
+                    "container.namespace_pull_containerdistribution",
+                    "container.namespace_push_containerdistribution",
+                    "container.namespace_change_containerdistribution",
+                    "container.namespace_view_containerpushrepository",
+                    "container.namespace_modify_content_containerpushrepository",
+                ],
+            },
+        ],
+    }
+}
+
+
+def set_pulp_container_access_policy(apps, schema_editor):
+    AccessPolicy = apps.get_model("core", "AccessPolicy")
+    print(AccessPolicy)
+    for view in viewsets:
+        policy, created = AccessPolicy.objects.update_or_create(
+            viewset_name=view, defaults={**viewsets[view], "customized": True})
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('galaxy', '0018_set_rate_limit_default'),
+        ('container', '0018_containerdistribution_description')
+    ]
+
+    operations = [
+        migrations.RunPython(
+            code=set_pulp_container_access_policy,
+        ),
+        migrations.CreateModel(
```

Comment:
Kind of want to suggest the model creation should be in a separate migration script, but :shrug: I don't think it matters.

Reply:
Creating the ContainerNamespace model will also create it's django_permissions right?  Is there any ordering issues with creating the model after the AccessPolicy item update?

The perms used in the policy are all really perms created with pulp_container model creation and already exist right?

Reply:
The ContainerNamespace is just getting created as a proxy model that we can use to add some extra lifecycle methods on. The actual ContainerNamespace is from pulp_container.

Reply:
:+1: